mmv1/third_party/tgc/imagewrapper.go

```diff
@@ -0,0 +1,23 @@
+package compute
+
+import (
+	tpg_services_compute "github.com/hashicorp/terraform-provider-google/google/services/compute"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+// ResolveImageWrapper is a wrapper around compute.ResolveImage and recover
+// from panic if client is il.
+func ResolveImageWrapper(c *transport_tpg.Config, project, name, userAgent string) (ret string, err error) {
```

Comment:
> However, compute_disk.go is still using ResolveImage, and since it's not a handwritten file, so cannot replace that function.

I think we need to figure out a way to call this `ResolveImage` so that it's used automatically by compute disk as well, since that's the original bug.

It looks like that might be easiest to do by completely removing `image.go`, making it export any functions TGC needs to reference, and extending this file to explicitly wrap all those functions (in most cases not changing their functionality).

If that would be excessive, it would also be possible to convert image.go to an erb file and exclude ResolveImage when rendering TGC - or use a different implementation - but that would potentially be more confusing.

attn @rileykarson in case you want to weigh in.

Reply:
I don't think TPG should require extensive modification due to a TGC issue. `image.go` requires online calls, I don't think it's unreasonable to have the same expectation in TFV.

Reply:
@rileykarson people in sandboxed environments sometimes need to be able to run TGC without network access. That said, we could totally decide to stop supporting that use case. Perhaps we should put this PR on hold until we can discuss further.

Reply:
My concern is, it will be having a larger impact, since the compute disk logic is generated from mmv1/templates/terraform/encoders/disk.erb, mmv1/templates/terraform/constants/disk.erb etc. 
They reference ResolveImage or other private functions within image.go. 
Check returned error logic will need to be baked into these erb files as well to cover the case for all kinds of compute disk case.